atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/store/AtlasDbTransactionStore.java

```diff
@@ -99,9 +113,22 @@ public Optional<WitnessedTransaction> readWrite(Consumer<InteractiveTransaction>
                     .build());
         } catch (SafeIllegalArgumentException e) {
             throw e;
+        } catch (KeyAlreadyExistsException e) {
+            Transaction transaction = transactionReference.get();
+            return Optional.of(MaybeWitnessedTransaction.builder()
+                    .startTimestamp(transaction.getTimestamp())
+                    .commitTimestamp(commitTimestampProvider
+                            .get()
+                            .getCommitTimestampOrThrowIfMaybeNotCommitted(transaction.getTimestamp()))
```

Comment:
not actionable: This might work strangely if the transaction task deliberately throws a `KeyAlreadyExistsException`, though I don't think we need to worry about that

Reply:
Later on, I made it the more general case of `TransactionCommitFailedException`, rather than just KAE, but still, a transaction could also throw this for funs. I think in this case we're ok, as we validate later on if it exists or not